src/Metrics/Metrics.hs

```diff
@@ -33,44 +32,29 @@ data ProjectMetrics = ProjectMetrics
   , projectMetricsMergeTrainSize :: Vector ProjectLabel Gauge
   }
 
-data MetricsOperationFree a
-  = MergeBranch a
-  | UpdateTrainSize Int a
-  deriving (Functor)
+data MetricsOperation :: Effect where
+  MergeBranch :: MetricsOperation m ()
+  UpdateTrainSize :: Int -> MetricsOperation m ()
 
-type MetricsOperation = Free MetricsOperationFree
+type instance DispatchOf MetricsOperation = 'Dynamic
 
-newtype LoggingMonitorT m a = LoggingMonitorT { runLoggingMonitorT :: LoggingT m a }
-                              deriving (Functor, Applicative, Monad, MonadIO, MonadLogger)
+increaseMergedPRTotal :: MetricsOperation :> es => Eff es ()
+increaseMergedPRTotal = send MergeBranch
 
-newtype NoMonitorT m a = NoMonitorT { runNoMonitorT :: NoLoggingT m a }
```

Comment:
The metrics were previously implemented on top of `LoggingT`. It turns out that the core class, `MonadMonitor`, is basically an alias of `MonadIO`, so the `LoggingT` is not necessary at all. This could just be implemented as an Effect on top of IO.